src/SDKs/Azure.Core/data-plane/Azure.Core/Request.cs

```diff
@@ -28,6 +28,18 @@ public virtual void AddHeader(string name, string value)
 
         public abstract bool TryGetHeader(string name, out string value);
 
+        public abstract bool TryGetHeaderValues(string name, out IEnumerable<string> values);
+
+        public abstract bool ContainsHeader(string name);
+
+        public virtual void SetHeader(string name, string value)
```

Comment:
What's the scenario for this?

Reply:
Most of them, adding another header value is rarely what you intend when setting a header, most of the time you want to override existing value with a new one.

Is illustrated by thousands of lines of generated code too: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/Microsoft.Azure.KeyVault/src/Generated/KeyVaultClient.cs#L342